src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2907,6 +2907,11 @@ internal string GetMessage(ITypeSymbol source, ITypeSymbol destination)
         /// </summary>
         public abstract IEnumerable<ISymbol> GetSymbolsWithName(Func<string, bool> predicate, SymbolFilter filter = SymbolFilter.TypeAndMember, CancellationToken cancellationToken = default(CancellationToken));
 
+        /// <summary>
+        /// Return source declaration symbols whose name matches the provided name
+        /// </summary>
+        internal abstract IEnumerable<ISymbol> GetSymbolsWithName(string name, SymbolFilter filter = SymbolFilter.TypeAndMember, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
i woudl like to make this public.  This would also be a valuable member for the FindReferencesEngine to have.  Currently we have to call into the member above this which is quite expensive.